### DISCUSSION TITLE
Suggest implement ready-to-use testing system

### DISCUSSION THREAD

**Discussion by TrungRueta** [antfu/vitesse#158] [Q&A]
Hi!, thank for you project helped developers integrate vitejs faster! As you see the most thing missing in vite eco-system for now is testing, still has not any office choice to setup testing, (jest | mocha ..). Do you think we can go forward a step to setup testing included in `vitesse` ?

Thank!

**Comment by antfu** [comment]
What kind of tests you are expecting? Unit tests or E2E tests?

**Comment by axe-me** [comment]
I actually just started a new project yesterday with this template, and right now adding jest to the project. I can help to add jest to the template with a PR if you think it's something good to have for this template.

**Comment by axe-me** [comment]
after spending some time on this, seems like making a vite transformer for jest is blocked by jest async transformer feature atm. see https://github.com/vitejs/vite/issues/1955

Without transform with vite, lots vite plugins that this template is using won't work properly.

As don't know when Jest27 gonna release, an alternative is use web test runner + mocha

**Comment by ChasLui** [comment]
I want to integrate jest and cypress
